### PR TITLE
feat(my-tickets): MyTicketsController + Provider (upcoming/past split)

### LINKED ISSUE #639: feat: MyTicketsController + Provider 생성
### 배경
내 티켓 화면의 비즈니스 로직 — upcoming/past 분리, 오늘 이벤트 감지.

**에픽**: #636 (내 티켓)
**의존성**: getMyTickets() Repository 메서드 (#637)
**기술 설계**: [`plan.md`](https://github.com/Mark-Yun/minglit/blob/dev/docs/features/my-tickets/plan.md)

### 수정 대상
| 파일 | 변경 내용 |
|------|----------|
| **신규** `apps/app_user/lib/src/features/my_tickets/logic/my_tickets_controller.dart` | Controller + Provider |

### 구현 가이드
1. `MyTicketsController` 생성 (Riverpod AsyncNotifier 또는 StateNotifier)
2. `getMyTickets()` 호출 후 `event.startTime` 기준으로 upcoming/past 분리
3. upcoming: startTime > now, ASC 정렬 (가까운 순)
4. past: startTime <= now, DESC 정렬 (최근 종료 순)
5. `todayEvent`: 오늘 시작하는 이벤트 중 가장 임박한 1개 (없으면 null)
6. PostgREST에서 관계 테이블 기준 정렬 불가 → 클라이언트 분리/정렬

### 검증 방법
- `flutter analyze` 통과
- 유닛 테스트: upcoming/past 분리, 정렬 순서, todayEvent 감지/null, 빈 리스트 처리

### PR DESCRIPTION
**Pull request by Mark-Yun**
## Summary
- Closes #639
- Add `MyTicketsController` with Riverpod `@riverpod` code generation
- Fetches active tickets (paid/approved) via `getMyTickets()` and splits into upcoming/past by event start time
- Detects `todayEvent` — the most imminent event starting today
- 7 unit tests covering: empty state, split logic, sort order (ASC upcoming, DESC past), today detection, null event handling

## 피처 파이프라인 히스토리
| 단계 | 이슈 | PR | 산출물 |
|------|------|-----|--------|
| Repository (getMyTickets) | #637 | (merged) | getMyTickets() |
| StatusBadge 추출 | #638 | (merged) | status_badge.dart |
| Controller + Provider | #639 | 현재 | my_tickets_controller.dart |

**피처 문서**: `docs/features/my-tickets/`

## Test plan
- [ ] `flutter analyze` passes
- [ ] Unit tests pass: `flutter test apps/app_user/test/src/features/my_tickets/logic/my_tickets_controller_test.dart`
- [ ] Verify upcoming sorted ascending (nearest first)
- [ ] Verify past sorted descending (most recent first)
- [ ] Verify todayEvent detection with today's events
- [ ] Verify empty state when user is logged out

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by Mark-Yun** [comment]
🤖 **TPM** CI 실패 분석

**실패 원인**: `flutter analyze` — `my_tickets_controller_test.dart`

| 라인 | 에러 | 수정 방향 |
|------|------|----------|
| L6 | `uri_does_not_exist` — `../../../../../utils/mocks.dart` 경로 없음 | mock 파일 경로 수정 또는 mock 파일 생성 |
| L34, L37 | `MockEventRepository` 미정의 | mock 클래스 정의 추가 (mockito `@GenerateMocks` 또는 수동 mock) |
| L61, L100, L137 | `MockUser` 미정의 | 동일 — mock 클래스 정의 필요 |

수정 후 같은 브랜치에 push하면 CI가 재실행됩니다.

File: apps/app_user/lib/src/features/my_tickets/logic/my_tickets_controller.dart [hunk 1]
```diff
@@ -30,67 +21,78 @@ class MyTicketsController extends _$MyTicketsController {
...
+  });
+
+  final List<EventApplication> upcoming;
+  final List<EventApplication> past;
+
+  /// The most imminent event starting today, or null.
+  final EventApplication? todayEvent;
+
+  bool get isEmpty => upcoming.isEmpty && past.isEmpty;
+}
```
[the remark this answers is not in the document]
Comment (Mark-Yun) [reply apps/app_user/lib/src/features/my_tickets/logic/my_tickets_controller.dart:99]: Fixed in 71f7a01e: MyTicketsState constructor now wraps lists with `List.unmodifiable()`. Removed `const` from constructor and adjusted the call site in `build()`.

File: apps/app_user/test/src/features/my_tickets/logic/my_tickets_controller_test.dart [hunk 2]
```diff
@@ -160,50 +169,65 @@ void main() {
...
+    test('detects todayEvent as most imminent event starting today', () async {
+      const userId = 'test_user';
+      final mockUser = MockUser();
+      when(() => mockUser.id).thenReturn(userId);
+
       final now = DateTime.now();
-      final todayStart = DateTime(now.year, now.month, now.day);
+      // Two events today: one sooner, one later
+      final todaySoon = DateTime(now.year, now.month, now.day, now.hour + 1);
+      final todayLate = DateTime(now.year, now.month, now.day, 23);
```
[the remark this answers is not in the document]
Comment (Mark-Yun) [reply apps/app_user/test/src/features/my_tickets/logic/my_tickets_controller_test.dart]: Fixed in 71f7a01e: Replaced `now.hour + 1` with fixed hour literals (22 and 23) to avoid midnight edge-case failures. These late-night hours ensure the events remain in the 'upcoming' bucket during normal CI run times.

**Comment by Mark-Yun** [comment]
🤖 Retrigger CI (commit mismatch)

File: apps/app_user/test/src/features/my_tickets/logic/my_tickets_controller_test.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mark-Yun) [reply apps/app_user/test/src/features/my_tickets/logic/my_tickets_controller_test.dart]: Addressed: replaced `now.add(Duration(minutes: 5/30))` with deterministic `DateTime(now.year, now.month, now.day, 22)` and `DateTime(now.year, now.month, now.day, 23)`. Guard clause now skips only if past 22:00 (CI runs during business hours), reducing the flaky window from ~30 minutes around midnight to a 2-hour off-hours window. Committed in a5e3d2cf.

File: apps/app_user/lib/src/features/my_tickets/logic/my_tickets_controller.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mark-Yun) [reply apps/app_user/lib/src/features/my_tickets/logic/my_tickets_controller.dart:99]: Already resolved: `List.unmodifiable()` wrapping was added in a previous commit (71f7a01e). The constructor stores immutable list views for both `upcoming` and `past` fields.
